common-app-lib/src/cmr/common_app/services/kms_lookup.clj

```diff
@@ -240,32 +242,46 @@
   "Takes a kms-index, the keyword scheme, and a short name and returns the full KMS hierarchy for
   that short name. Returns nil if a keyword is not found. Comparison is made case insensitively."
   [context keyword-scheme short-name]
-  (when-not (:ignore-kms-keywords context)
-    (let [short-name-cache (hash-cache/context->cache context kms-short-name-cache-key)
-          [tm keywords] (util/time-execution (hash-cache/get-value short-name-cache kms-short-name-cache-key keyword-scheme))
-          _ (rl-util/log-redis-read-complete "lookup-by-short-name" kms-short-name-cache-key tm)
-          keywords (if keywords
-                     keywords
-                     (let [_ (load-cache-if-necessary context short-name-cache kms-short-name-cache-key)
-                           [tm kwords] (util/time-execution (hash-cache/get-value short-name-cache kms-short-name-cache-key keyword-scheme))]
-                       (rl-util/log-redis-read-complete "lookup-by-short-name" kms-short-name-cache-key tm)
-                       kwords))]
-      (get keywords (util/safe-lowercase short-name)))))
+  (try
+    (when-not (:ignore-kms-keywords context)
+      (let [short-name-cache (hash-cache/context->cache context kms-short-name-cache-key)
+            [tm keywords] (util/time-execution (hash-cache/get-value short-name-cache kms-short-name-cache-key keyword-scheme))
+            _ (rl-util/log-redis-read-complete "lookup-by-short-name" kms-short-name-cache-key tm)
+            keywords (if keywords
+                       keywords
+                       (let [_ (load-cache-if-necessary context short-name-cache kms-short-name-cache-key)
+                             [tm kwords] (util/time-execution (hash-cache/get-value short-name-cache kms-short-name-cache-key keyword-scheme))]
+                         (rl-util/log-redis-read-complete "lookup-by-short-name" kms-short-name-cache-key tm)
+                         kwords))]
+        (get keywords (util/safe-lowercase short-name))))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "lookup-by-short-name found redis carmine exception. Will return nil result." e)
+          nil)
```

Comment:
Remove the (do and the nil)
(error returns nil.

Please do this in the kms_lookup namespace as well.